alf/utils/losses.py

```diff
@@ -714,26 +714,34 @@ class BipartiteMatchingLoss(object):
     """
 
     def __init__(self,
-                 pair_loss_fn: Callable = torch.cdist,
+                 pair_loss_fn: alf.nest.Nest = torch.cdist,
+                 loss_weights: alf.nest.Nest = 1.,
                  reduction: str = 'mean',
                  name: str = "BipartiteMatchingLoss"):
         """
         Args:
-            pair_loss_fn: the pairwise matching loss function. It should take
-                two sets of length ``N`` and output a cost matrix of shape ``[N,N]``.
+            pair_loss_fn: a nest of pairwise matching loss functions. Each element
```

Comment:
I don't see the necessity of having nested pair_loss_fn in this module. Why can't you let pair_loss_fn to handle nested input. And perhaps we can even go without pair_loss_fn. Just use cost_mat as input.